src/triggered/CommunicationMapping.ts

```diff
@@ -0,0 +1,125 @@
+import { LitElement } from 'lit-element';
+
+import { getDataSink, newWizardEvent } from '../foundation.js';
+
+import { commMappingWizard } from './wizards.js';
+
+export type ControlBlockType =
+  | 'ReportControl'
+  | 'LogControl'
+  | 'GSEControl'
+  | 'SampledValuesControl';
+
+type ControlBlockConnectionType = 'IEDName' | 'ClientLN';
+
+const cbConnection: Record<ControlBlockType, ControlBlockConnectionType> = {
+  ReportControl: 'ClientLN',
+  LogControl: 'ClientLN',
+  GSEControl: 'IEDName',
+  SampledValuesControl: 'IEDName',
+};
+
+export interface Connection {
+  source: Element;
+  data?: Element;
+  sink: Element;
+}
+
+export function getConnectionIndexOf(
+  list: Connection[],
+  match: Connection
+): number {
+  for (let i = 0; list.length; i++)
+    if (
+      list[i].source.closest('IED')?.getAttribute('name') ===
+        match.source.closest('IED')?.getAttribute('name') &&
+      list[i].source.getAttribute('name') ===
+        match.source.getAttribute('name') &&
+      list[i].sink.getAttribute('iedName') ===
+        match.sink.getAttribute('iedName')
+    )
+      return i;
+
+  return -1;
+}
+
+export function getConnection(
+  connections: Connection[],
+  item: Connection
+): Connection[] {
+  return connections.filter(
+    match => match.source === item.source && match.sink === match.sink
+  );
+}
+
+/**
+ * @returns array of - control block - Connection's
+ * Control block connection are connections defined within the control block
+ * by the element `ClientLN` for `ReportControl` and `LogControl` and
+ * by the element `IEDName` for `GSEControl` and `SampledValueControl`
+ */
+export function getControlBlockConnection(
+  root: Document | Element,
+  cbType: ControlBlockType
+): Connection[] {
+  return Array.from(root.querySelectorAll(cbType))
+    .filter(controlBlock => !controlBlock.closest('Private'))
+    .flatMap(controlBlock => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/openscd/open-scd/pullRequest?prid=7203429)
```suggestion
    .flatMap((controlBlock) => {
```